src/capi/atom.cpp

```diff
@@ -200,6 +200,28 @@ extern "C" CHFL_PROPERTY* chfl_atom_get_property(const CHFL_ATOM* const atom, co
     return nullptr;
 }
 
+extern "C" chfl_status chfl_atom_properties_names(const CHFL_ATOM* atom, uint64_t* count, const char*** names) {
+    CHECK_POINTER(atom);
+    auto opt_atom_properties_begin = atom->properties_begin();
```

Comment:
The whole function body needs to be wrapped in a `CHFL_ERROR_CATCH` block as any line might throw an exception.